### PR TITLE
nx-read-only RSC-396

### PR DESCRIPTION
**Pull request by glenhunter**
https://issues.sonatype.org/browse/RSC-396

### CODE REVIEW

File: gallery/src/styles/NxReadOnly/NxReadOnlyDlExample.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    the terms of the Eclipse Public License 2.0 which accompanies this
+    distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+
+-->
+<dl class="nx-read-only">
+  <dt class="nx-read-only__label">
+    Label that describes the data that will appear below
+  </dt>
+  <dd class="nx-read-only__data">
+    Data - found security vulnerability CVE-2020-6230 with severity < 10 (severity = 7.2)
```
Comment (rpokorny) [inline gallery/src/styles/NxReadOnly/NxReadOnlyDlExample.html:14]: Don't you need to escape the `<` here?

File: gallery/src/styles/NxReadOnly/NxReadOnlyDlExample.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<!--
+
+    Copyright (c) 2019-present Sonatype, Inc.
+    This program and the accompanying materials are made available under
+    the terms of the Eclipse Public License 2.0 which accompanies this
+    distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+
+-->
+<dl class="nx-read-only">
+  <dt class="nx-read-only__label">
```
Comment (rpokorny) [inline gallery/src/styles/NxReadOnly/NxReadOnlyDlExample.html:10]: This example only has one `dt`.  I think it needs several `dt`/`dd` groups to show the spacing between them.

File: gallery/src/styles/NxReadOnly/NxReadOnlyDlExample.html [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  </dt>
+  <dd class="nx-read-only__data">
+    Data - found security vulnerability CVE-2020-6230 with severity < 10 (severity = 7.2)
+  </dd>
+  <dd class="nx-read-only__data">
+    Coordinates were com.orientechnologies : orientdb-core : 2.2.36 (do not match org.sonatype.nexus.assemblies :
+    nexus-bundle-template : * : * : *)
+  </dd>
+  <dd class="nx-read-only__data">
+    Found security vulnerability CVE-2020-6230 with severity >= 7 (severity = 7.2)
```
Comment (rpokorny) [inline gallery/src/styles/NxReadOnly/NxReadOnlyDlExample.html:21]:
```suggestion
    Found security vulnerability CVE-2020-6230 with severity &gt;= 7 (severity = 7.2)
```

File: gallery/src/styles/NxReadOnly/NxReadOnlyPage.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import { GalleryDescriptionTile, GalleryExampleTile } from '../../gallery-components/GalleryTiles';
+
+const nxReadOnlyDlExampleCode = require('!!raw-loader!./NxReadOnlyDlExample.html').default;
+
+const NxReadOnlyPage = () => {
+  return (
+    <>
+      <GalleryDescriptionTile>
+        <p className="nx-p">
+          <code className="nx-code">.nx-read-only</code> is a design pattern for when you wish to display data in a read only form. It's basic structure is a label and one or more data points.
```
Comment (rpokorny) [inline gallery/src/styles/NxReadOnly/NxReadOnlyPage.tsx:18]: Line too long

File: gallery/src/styles/NxReadOnly/NxReadOnlyPage.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+            <tr className="nx-table-row">
+              <td className="nx-cell"><code className="nx-code">.nx-read-only</code></td>
+              <td className="nx-cell">Top level</td>
+              <td className="nx-cell">
+                This is the basic wrapper class. It is applied to the <code className="nx-code">&lt;dl&gt;</code>.
+              </td>
+            </tr>
+            <tr className="nx-table-row">
+              <td className="nx-cell"><code className="nx-code">.nx-read-only__label</code></td>
+              <td className="nx-cell">Element</td>
```
Comment (rpokorny) [inline gallery/src/styles/NxReadOnly/NxReadOnlyPage.tsx:42]: Probably should say which element (the `dt`).  Same in the `.nx-read-only__data` section below.

File: gallery/src/styles/NxReadOnly/NxReadOnlyPage.tsx (hunk 4 above)
Comment (glenhunter) [reply gallery/src/styles/NxReadOnly/NxReadOnlyPage.tsx]: > Line too long

Fixed here: https://github.com/sonatype/sonatype-react-shared-components/pull/243/commits/e352234a25ff61e98f1756f843d2ca0afc123bfa

File: gallery/src/styles/NxReadOnly/NxReadOnlyPage.tsx (hunk 5 above)
Comment (glenhunter) [reply gallery/src/styles/NxReadOnly/NxReadOnlyPage.tsx:43]: > Probably should say which element (the `dt`). Same in the `.nx-read-only__data` section below.

In both cases it says which element in the descriptive text that follows

File: gallery/src/styles/NxReadOnly/NxReadOnlyDlExample.html (hunk 2 above)
Comment (glenhunter) [reply gallery/src/styles/NxReadOnly/NxReadOnlyDlExample.html:10]: > This example only has one `dt`. I think it needs several `dt`/`dd` groups to show the spacing between them.

I flashed out the examples here: https://github.com/sonatype/sonatype-react-shared-components/pull/243/commits/f03dab91c22292c7ee5753fca1e53ad3238c23bc

File: gallery/src/styles/NxReadOnly/NxReadOnlyPage.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+      <GalleryDescriptionTile>
+        <p className="nx-p">
+          <code className="nx-code">.nx-read-only</code> is a design pattern for when you wish to display data in a
+          read only form. It's basic structure is a label and one or more data points.
+        </p>
+        <p className="nx-p">
+          <code className="nx-code">.nx-read-only</code> is layed out using a description list
+          (<code className="nx-code">&lt;dl&gt;</code>) as the container.
+        </p>
+        <table className="nx-table nx-table--gallery-props">
```
Comment (rpokorny) [inline gallery/src/styles/NxReadOnly/NxReadOnlyPage.tsx:25]: I've been trying to use `NxTable` on new pages.  At the very least, `.nx-table--gallery-props` no longer does anything as far as I'm aware.

File: lib/src/base-styles/_nx-read-only.scss [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  margin: 0 0 $nx-spacing-xl 0;
+}
+
+.nx-read-only__label {
+  @include bold();
+  margin: $nx-spacing-xl 0 $nx-spacing-md 0;
+
+  &:first-child {
+    margin-top: 0;
+  }
```
Comment (rpokorny) [inline lib/src/base-styles/_nx-read-only.scss:22]: Redundant with `container-vertical` on `.nx-read-only`
```suggestion
```

File: lib/src/base-styles/_nx-read-only.scss [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+.nx-read-only__data {
+  margin: 0 0 $nx-spacing-xs 0;
+  padding: 0;
```
Comment (rpokorny) [inline lib/src/base-styles/_nx-read-only.scss:27]: This doesn't seem like it does anything

File: lib/src/base-styles/_nx-read-only.scss [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+ @import '../scss-shared/nx-variables';
+ @import '../scss-shared/nx-text-helpers';
+ @import '../scss-shared/nx-container-helpers';
+
+.nx-read-only {
+  @include container-vertical;
+  margin: 0 0 $nx-spacing-xl 0;
```
Comment (rpokorny) [inline lib/src/base-styles/_nx-read-only.scss:13]: No top margin?

File: lib/src/base-styles/_nx-read-only.scss (hunk 7 above)
Comment (glenhunter) [reply lib/src/base-styles/_nx-read-only.scss]: > Redundant with `container-vertical` on `.nx-read-only`

https://github.com/sonatype/sonatype-react-shared-components/pull/243/commits/e9eb2959db0de3996115e2a74ef5ac4df1174a58

File: lib/src/base-styles/_nx-read-only.scss (hunk 8 above)
Comment (glenhunter) [reply lib/src/base-styles/_nx-read-only.scss]: > This doesn't seem like it does anything

https://github.com/sonatype/sonatype-react-shared-components/pull/243/commits/e9eb2959db0de3996115e2a74ef5ac4df1174a58

File: lib/src/base-styles/_nx-read-only.scss (hunk 9 above)
Comment (glenhunter) [reply lib/src/base-styles/_nx-read-only.scss:13]: > No top margin?

I used the exact same settings as we use for `.nx-form-group` which does not specify top margin.

File: gallery/src/styles/NxReadOnly/NxReadOnlyPage.tsx (hunk 6 above)
Comment (glenhunter) [reply gallery/src/styles/NxReadOnly/NxReadOnlyPage.tsx]: > I've been trying to use `NxTable` on new pages. At the very least, `.nx-table--gallery-props` no longer does anything as far as I'm aware.

Ah right: https://github.com/sonatype/sonatype-react-shared-components/pull/243/commits/9ad4504c99bb00bebc4b42ac5b3a468f96f8b540

File: gallery/src/styles/NxReadOnly/NxReadOnlyPage.tsx [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+const nxReadOnlyDlExampleCode = require('!!raw-loader!./NxReadOnlyDlExample.html').default;
+
+const NxReadOnlyPage = () => {
+  return (
+    <>
+      <GalleryDescriptionTile>
+        <p className="nx-p">
+          <code className="nx-code">.nx-read-only</code> is a design pattern for when you wish to display data in a
+          read only form. It's basic structure is a label and one or more data points.
```
Comment (rpokorny) [inline gallery/src/styles/NxReadOnly/NxReadOnlyPage.tsx:19]:
```suggestion
          <code className="nx-code">.nx-read-only</code> is a design pattern for when you wish to display read-only data in a form. Its basic structure is a label and one or more data points.
```

**Review by rpokorny — APPROVED** [review]

**Pull request merged**